### PR TITLE
[infrastructure] Added spotless maven code formatter config

### PR DESCRIPTION
**Pull request by seime**
See https://github.com/openhab/openhab-core/issues/935

Signed-off-by: Arne Seime <arne.seime@gmail.com>

### CODE REVIEW

**Comment by seime** [comment]
Depends on https://github.com/openhab/openhab-core/pull/1101

File: pom.xml [hunk 1]
```diff
@@ -521,6 +523,75 @@ Import-Package: \\
...
           </executions>
         </plugin>
+        <plugin>
+          <groupId>com.diffplug.spotless</groupId>
+          <artifactId>spotless-maven-plugin</artifactId>
+          <version>${spotless.version}</version>
+          <configuration>
+            <java>
+              <eclipse>
+                <file>openhab_codestyle.xml</file>
```
Comment (hakan42) [inline pom.xml]: Where is this openhab_codestyle.xml file? I can't find it in the current branch or in the master...

Comment (seime) [reply pom.xml]: It is located here https://github.com/seime/openhab-core/tree/spotless/tools/codestyle/src/main/resources (PR to be merged first) and loaded as a classpath entry by the plugin.

Comment (seime) [reply pom.xml]: Correct

File: pom.xml [hunk 2]
```diff
@@ -521,6 +523,75 @@ Import-Package: \\
...
+              <importOrder>
+                <file>openhab.importorder</file>
+              </importOrder>
+              <endWithNewline />
+            </java>
+            <formats>
+              <format>
+                <!-- *.xml -->
+                <includes>
+                  <include>src/**/*.xml</include>
```
[the remark this answers is not in the document]
Comment (seime) [reply pom.xml]: It is probably wise to have this in mind when we start reformatting the various projects/bundles. For most files my gut feeling is that reformatting won't hurt. But there could be cases where reformatting will destroy test data. 
Creating an example of how to override which XML files to reformat could be an idea?

**Comment by seime** [comment]
Depends on https://github.com/openhab/openhab2-addons/pull/6338

**Pull request merged**